tools/project_manager/show_word_context.php

```diff
@@ -28,115 +27,93 @@
 
 // get the correct layout
 $userSettings =& Settings::get_Settings($pguser);
-// if not set gives LAYOUT_HORIZ
-$default_layout =  $userSettings->get_value("show_word_context_layout", LAYOUT_HORIZ);
-
-$layout_choices = array(LAYOUT_HORIZ, LAYOUT_VERT);
-$layout = get_enumerated_param($_GET, 'layout', $default_layout, $layout_choices);
-if($layout != $default_layout)
-{
-    $userSettings->set_value("show_word_context_layout", $layout);
-}
+// if not set gives "horizontal"
+$default_layout =  $userSettings->get_value("show_word_context_layout", "horizontal");
 
 $wordInstances =  get_integer_param($_GET, 'wordInstances', 20, 0, MAX_WORD_INSTANCES);
 
-// $frame determines which frame we're operating from
-// 'master' - we're the master frame
-//  'left'  - we're the left frame with the text
-// 'right'  - we're the right frame for the image
-$frame = get_enumerated_param($_GET, 'frame', 'master', array('master', 'left', 'right'));
-
-if($frame=="master") {
-    slim_header_frameset(_("Word Context"));
-    if($layout == LAYOUT_HORIZ) $frameSpec='rows="30%,70%"';
-    else $frameSpec='cols="30%,70%"';
-?>
-<frameset <?php echo $frameSpec; ?>>
-<frame name="worddetailframe" src="show_word_context.php?projectid=<?php echo $projectid; ?>&amp;word=<?php echo $encWord; ?>&amp;wordInstances=<?php echo $wordInstances; ?>&amp;frame=left">
-<frame name="imageframe" src="show_word_context.php?projectid=<?php echo $projectid; ?>&amp;word=<?php echo $encWord; ?>&amp;wordInstances=<?php echo $wordInstances; ?>&amp;frame=right">
-</frameset>
-<noframes>
-<?php echo _("Your browser currently does not display frames!"); ?>
-</noframes>
-<?php
-    exit;
+$details = json_encode([
+    "layout" => $default_layout,
+    "projectid" => $projectid,
+    'storageKeyLayout' => 'show_word_context_layout',
+]);
+
+$header_args = [
+    "js_files" => [
+        "$code_url/scripts/splitControl.js",
+        "$code_url/scripts/page_browse.js",
+        "./show_word_context.js",
+    ],
+    "js_data" => get_proofreading_interface_data_js() . "
+        var showWordContext = $details;",
+
+    "body_attributes" => 'class="no-margin overflow-hidden" style="height: 100vh; width: 100vw"',
+];
+
+slim_header(_("Word Context"), $header_args);
+echo "<div id='show_word_context_container' style='flex: auto;width: 100%;height: 100%'>";
+echo "<div class='overflow-auto'>";
+echo "<div style='padding: 0.5em'>";
+
+$project_name = get_project_name($projectid);
+// TRANSLATORS: %1$s is a word, %2$s is the project name.
+echo "<h2>", sprintf(_("Context for '%1\$s' in %2\$s"), $word, $project_name), "</h2>";
+
+echo "<p>";
+echo "<a href='javascript:void(0)' id='h_v_switch'></a>";
```

Comment:
I think we should use `href='#'` for these links-that-aren't-a-link, as that appears to be more standard-compliant: https://ourcodeworld.com/articles/read/209/should-i-use-hash-or-javascript-void-0-on-the-href-attribute-of-my-empty-links